### DISCUSSION TITLE
`TimelineZoom` should be a local value

### DISCUSSION THREAD

**Discussion by cihe13375** [ppy/osu#27550] [Ideas]
Currently the `TimelineZoom` (determines how compact the timeline is in the editor) is encoded in the .osu file, and is considered a part of the map. This doesn't really make sense, since it
1) does not affect playing experience in any observable way;
2) is more of a mapper's own perference rather than actually related to the song/map.

The problem becomes apparent when checking some maps that have extreme `TimelineZoom` values (like https://osu.ppy.sh/beatmapsets/1976137#osu/4101562), where one have to either (1) manually zoom back every time entering the editor, or (2) save the new zoom locally and lose its online status.

The problem may be solved in two ways:
1) allow local override of `TimelineZoom` without losing the online status. Maybe the new zoom value is saved in some local database and is prioritized over the value in .osu file, while the .osu file is untouched.
2) deprecate `TimelineZoom` and move it to a setting in the game, either global or per-beatmap. Assuming a global setting, it could be based on either
(a) the distance per beat (i.e. an 1/1 gap always has a fixed length in the timeline, regardless of bpm), or
(b) the distance per second (i.e. every second always has a fixed length in the timeline, regardless of the beats included).

**Comment by peppy** [comment]
Probably.
